cmd/podman/inspect.go

```diff
@@ -71,81 +71,103 @@ func inspectCmd(c *cli.Context) error {
-			}
-			data, err = getCtrInspectInfo(ctr, libpodInspectData)
-			if err != nil {
-				return errors.Wrapf(err, "error parsing container data %q", ctr.ID)
-			}
+			return err
 		}
+		args = append(args, lc.ID())
+		inspectType = inspectTypeContainer
 	}
 
+	inspectedObjects, iterateErr := iterateInput(c, args, runtime, inspectType)
+
 	var out formats.Writer
 	if outputFormat != "" && outputFormat != formats.JSONString {
 		//template
-		out = formats.StdoutTemplate{Output: data, Template: outputFormat}
+		out = formats.StdoutTemplateArray{Output: inspectedObjects, Template: outputFormat}
 	} else {
 		// default is json output
-		out = formats.JSONStruct{Output: data}
+		out = formats.JSONStructArray{Output: inspectedObjects}
 	}
 
 	formats.Writer(out).Out()
-	return nil
+	return iterateErr
+}
+
+// func iterateInput iterates the images|containers the user has requested and returns the inspect data and error
+func iterateInput(c *cli.Context, args []string, runtime *libpod.Runtime, inspectType string) ([]interface{}, error) {
+	var (
+		data           interface{}
+		inspectedItems []interface{}
+		inspectError   error
+	)
+
+	for _, input := range args {
+		switch inspectType {
+		case inspectTypeContainer:
+			ctr, err := runtime.LookupContainer(input)
+			if err != nil {
+				inspectError = errors.Wrapf(err, "error looking up container %q", input)
+				break
+			}
+			libpodInspectData, err := ctr.Inspect(c.Bool("size"))
+			if err != nil {
+				inspectError = errors.Wrapf(err, "error getting libpod container inspect data %q", ctr.ID)
+				break
+			}
+			data, err = getCtrInspectInfo(ctr, libpodInspectData)
+			if err != nil {
+				inspectError = errors.Wrapf(err, "error parsing container data %q", ctr.ID())
+				break
+			}
+		case inspectTypeImage:
+			image, err := runtime.GetImage(input)
+			if err != nil {
+				inspectError = errors.Wrapf(err, "error getting image %q", input)
+				break
+			}
+			data, err = runtime.GetImageInspectInfo(*image)
+			if err != nil {
+				inspectError = errors.Wrapf(err, "error parsing image data %q", image.ID)
+				break
+			}
+		case inspectAll:
+			ctr, err := runtime.LookupContainer(input)
+			if err != nil {
+				image, err := runtime.GetImage(input)
+				if err != nil {
+					inspectError = errors.Wrapf(err, "error getting image %q", input)
+					break
+				}
+				data, err = runtime.GetImageInspectInfo(*image)
+				if err != nil {
+					inspectError = errors.Wrapf(err, "error parsing image data %q", image.ID)
+					break
+				}
+			} else {
+				libpodInspectData, err := ctr.Inspect(c.Bool("size"))
+				if err != nil {
+					inspectError = errors.Wrapf(err, "error getting libpod container inspect data %q", ctr.ID)
+					break
+				}
+				data, err = getCtrInspectInfo(ctr, libpodInspectData)
+				if err != nil {
+					inspectError = errors.Wrapf(err, "error parsing container data %q", ctr.ID)
+					break
+				}
+			}
+		}
+		if inspectError == nil {
```

Comment:
Should we log inspectError if it's not nil here?

Reply:
theoretically i dont think the conditional is needed because the error handling is occurring in the switch case statements. i put it there for completeness but could be swayed to remove it.

Reply:
Eh, probably fine as is.